auth/authzn.proto

```diff
@@ -0,0 +1,179 @@
+//
+// Copyright 2020 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+syntax = "proto3";
+
+import "github.com/openconfig/gnoi/types/types.proto";
+
+// This package is a work in progress. It may change in the future. :-)
+package gnoi.authzn;
+
+option (types.gnoi_version) = "0.1.0";
+
+// Service Authzn supplies the target (i.e., network device, server end of the
+// gRPC connection) with a definition of an authentication and authorization
+// policy that should be used for the respective (authn/authz) of the relevant
+// service. 
+//
+// It is expected that when authorization and authentication details are
+// supplied via the gNOI Authzn service, this config cannot be manipulated
+// through other interfaces, such that a restriction on the gNOI.authzn RPC
+// itself restricts the set of users that are able to manipulate the auth[zn]
+// configuration of the end device.
+//
+// In order to ensure that the auth[zn] configuration supplied in the service
```

Comment:
Nit on auth[zn] ... logically authentication usually comes before authorization but these words' ordering isn't consistent in this text.  Also we often use "authn" and "authz" internally but these acronyms should probably be spelt out a little more.

Reply:
I was struggling with a name here that made sense to combine the two -- happy to use something different if it makes sense (`authnz` is consistent with that ordering if it's commonly used to mean this).

Once we've agreed here I'll make things consistent and spell it out - authn and authz are pretty common (used in IETF documents fairly regularly), so I don't think we have to worry too much about them being unclear.